### PR TITLE
PARAM.SFO loader improvement

### PR DESCRIPTION
**Pull request by elad335**

### CODE REVIEW

File: rpcs3/Loader/PSF.cpp [hunk 1]
```diff
@@ -246,6 +246,15 @@ namespace psf
 			}
 		}
 
+		const auto cat = get_string(pair.sfo, "CATEGORY", "123");
```
Comment (Megamouse) [inline rpcs3/Loader/PSF.cpp:249]: can just leave it empty

File: rpcs3/Emu/Cell/Modules/cellGame.cpp [hunk 2]
```diff
@@ -189,7 +189,7 @@ error_code cellHddGameCheck(ppu_thread& ppu, u32 version, vm::cptr<char> dirName
 	if (!new_data)
 	{
 		const auto cat = psf::get_string(sfo, "CATEGORY", "");
-		if (cat != "HG"sv && cat != "AM"sv && cat != "AP"sv && cat != "AT"sv && cat != "AV"sv && cat != "BV"sv && cat != "WT"sv)
+		if (cat.ends_with("D") || cat == "DG") // Checks againt game data and disc game
```
Comment (Megamouse) [inline rpcs3/Emu/Cell/Modules/cellGame.cpp:192]: No sv anymore?
```suggestion
		if (cat.ends_with("D") || cat == "DG" || cat == "MS") // Checks against game data and disc game
```

File: rpcs3/Emu/Cell/Modules/cellGame.cpp [hunk 3]
```diff
@@ -594,7 +594,7 @@ error_code cellGameDataCheck(u32 type, vm::cptr<char> dirName, vm::ptr<CellGameC
 	{
 		switch (type)
 		{
-		case CELL_GAME_GAMETYPE_HDD: return cat != "HG"sv && cat != "AM"sv && cat != "AP"sv && cat != "AT"sv && cat != "AV"sv && cat != "BV"sv && cat != "WT"sv;
+		case CELL_GAME_GAMETYPE_HDD: return cat.ends_with("D") || cat == "DG"; // Checks againt game data and disc game
```
Comment (Megamouse) [inline rpcs3/Emu/Cell/Modules/cellGame.cpp:597]: No sv anymore?
```suggestion
		case CELL_GAME_GAMETYPE_HDD: return cat.ends_with("D") || cat == "DG" || cat == "MS"; // Checks against game data and disc game
```

Comment (elad335) [reply rpcs3/Emu/Cell/Modules/cellGame.cpp]: Readded.

File: rpcs3/Loader/PSF.h [hunk 4]
```diff
@@ -108,4 +108,10 @@ namespace psf
 	{
 		return {format::array, max_size, value};
 	}
+
+	// Checks if of HDD catgeory (assumes a valid category is being passed)
+	constexpr bool is_cat_hdd(std::string_view cat)
+	{
+		return cat[1] != 'D' && cat != "DG"sv && cat != "MS"sv;
```
Comment (Megamouse) [inline rpcs3/Loader/PSF.h:115]: what if cat size is < 2 ?

Comment (Megamouse) [reply rpcs3/Loader/PSF.h:115]: I don't care if it should be fine, it's standard procedure to check for such things

Comment (elad335) [reply rpcs3/Loader/PSF.h]: Okay.

**Comment by Megamouse** [comment]
I think it's fine now.
Although I'd like to some testers to check first

**Pull request merged**